### PR TITLE
[AIRFLOW-4823] modified the source to have create_default_connections flag in config

### PR DESCRIPTION
**Pull request by Jagankmr**
… to create default connections by airflow

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by Jagankmr** [comment]
Updated the default_airflow.cfg to have a flag "create_default_connection" the same variable will be handled in airflow\airflow\utils\db.py file to create or not to create default connections while the airflow is deployed

**Review by ashb — CHANGES_REQUESTED** [review]
I don't think this should be a config option, but a flag that is passed in via the CLI.

And the tutorial/getting started guide will need to be updated to pass this flag when doing running init db

**Comment by Jagankmr** [comment]
@ashb - Changes implemented as per your comments

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -1720,6 +1720,9 @@ class CLIFactory:
             "Do not prompt to confirm reset. Use with care!",
             "store_true",
             default=False),
+        'default_connection': Arg(
+            ("-d", "--default_connection"), "Flag to create airflow default connections",
+            default=True),
```
Comment (ashb) [inline airflow/bin/cli.py:1725]: If this is true by default can it be turned off?

Comment (Jagankmr) [reply airflow/bin/cli.py:1725]: if we call cli like airflow initdb --default_connection False and if we just call airflow initdb will take True by default

Comment (ashb) [reply airflow/bin/cli.py:1725]: More normal would be `--no-default-connections` (a separate flag to turn it off/on)

Comment (Jagankmr3) [reply airflow/bin/cli.py]: no need i guess, --default_connection is enough why we need two variables here

Comment (Jagankmr) [reply airflow/bin/cli.py:1725]: what me to change the variable name from --default-connection to --no-default-connections?

**Comment by edgarrmondragon** [comment]
@Jagankmr are you still working on this?

File: airflow/bin/cli.py (hunk 1 above)
Comment (mik-laj) [reply airflow/bin/cli.py:1725]: I agree with @ashb. It seems more intuitive to me.

**Comment by Jagankmr** [comment]
> @Jagankmr are you still working on this?
@mrfunnyshoes am not doing any progress here.. i made changes but it is failing in CI testing. It would be great if someone help me on this.

**Pull request closed without merging**